crates/core/src/client/inner/logging.rs

```diff
@@ -0,0 +1,56 @@
+use std::{io::Write, sync::Once};
+
+use env_logger::{Builder, Env};
+use log::LevelFilter;
+
+use super::super::config::LogLevel;
+
+static INIT_LOG: Once = Once::new();
+
+impl From<LogLevel> for LevelFilter {
+    fn from(level: LogLevel) -> Self {
+        match level {
+            LogLevel::Trace => LevelFilter::Trace,
+            LogLevel::Debug => LevelFilter::Debug,
+            LogLevel::Info => LevelFilter::Info,
+            LogLevel::Warn => LevelFilter::Warn,
+            LogLevel::Error => LevelFilter::Error,
+        }
+    }
+}
+
+pub fn init_log(level: LogLevel) {
+    INIT_LOG.call_once(|| {
+        let env = Env::default();
+        let mut builder = Builder::from_env(env);
+        builder
+            .format(|buf, record| {
+                if record.level() == log::Level::Error {
+                    writeln!(
+                        buf,
+                        "[{}] {} {}:{} - {}",
+                        record.level(),
+                        record.target(),
+                        record.file().unwrap_or("unknown"),
+                        record.line().unwrap_or(0),
+                        record.args()
+                    )
+                } else {
+                    writeln!(
+                        buf,
+                        "[{}] {} - {}",
+                        record.level(),
+                        record.target(),
+                        record.args()
+                    )
+                }
+            })
+            .filter(None, level.into())
+            .try_init()
+            .expect("LOG INITIALIZATION FAILED");
+    });
+}
```

Comment:
Hmm. Does env logger work on the various platforms? Like, debugging on device kinda requires using [`oslog`](https://crates.io/crates/oslog). I feel like there was an android one (that I never got working).

Reply:
This is just a stub for now. I am going make this module way more configurable, but for testing purposes env log is fine. in the end it will be a multi-log with an env log and platform specific log.

Reply:
The other goal here - each platform will want something akin to a developer console. Providing a callback interface for appending to an application developer debug log which can filter out core events, present warnings and diagnostics, is an goal.